docs/design/Education-organization-Endpoints.md

```diff
@@ -0,0 +1,153 @@
+basis.
+
+## Features
+
+* **REST API Endpoints:**
+  * `GET /v2/educationorganizations` or `GET /v1/educationorganizations` -
+    Returns all education organizations from all instances
+  * `GET /v2/educationorganizations/{instanceId}` or
+    `GET /v1/educationorganizations/{instanceId}` - Returns education organizations
+    for a specific instance
+
+* **Automatic Data Refresh:**
+  * Quartz.NET scheduled job runs every 6 hours by default
+  * No manual refresh endpoint - data is refreshed automatically via background
+    job
+
+* **Cross-Database Support:**
+  * Works with both SQL Server and PostgreSQL
+  * Uses C# service layer for dynamic database connections and efficient data
+    refresh
+
+## Database Schema
+
+### Tables
+
+* `adminapi.EducationOrganizations` - Stores consolidated education organization
+  data
+
+  ```sql
+  CREATE TABLE [adminapi].[EducationOrganizations] (
+        [Id] INT IDENTITY(1,1) NOT NULL,
+        [InstanceId] INT NOT NULL,
+        [InstanceName] NVARCHAR(100) NOT NULL,
+        [EducationOrganizationId] INT NOT NULL,
+        [NameOfInstitution] NVARCHAR(75) NOT NULL,
+        [ShortNameOfInstitution] NVARCHAR(75) NULL,
+        [Discriminator] NVARCHAR(128) NOT NULL,
+        [ParentId] INT NULL,
+        [OdsDatabaseName] NVARCHAR(255) NULL,
+        [LastRefreshed] DATETIME2 NOT NULL DEFAULT GETUTCDATE(),
+        [LastModifiedDate] DATETIME2 NULL,
+        CONSTRAINT [PK_EducationOrganizations] PRIMARY KEY ([Id])
+    );
+  ```
+
+## API Usage Examples
+
+### Get All Education Organizations
+
+```http
+GET /v2/educationorganizations
+Authorization: Bearer <token>
+```
+
+### Get Education Organizations for Specific Instance
+
+```http
+GET /v2/educationorganizations/123
+Authorization: Bearer <token>
+```
+
+## Configuration
+
+### Quartz.NET Job Scheduling
+
+Add to `appsettings.json`:
+
+```json
+{
+   "AppSettings": {
+   "EducationOrgsRefreshIntervalInHours": 6
+   }
+}
+```
+
+### Database Connection
+
+The system uses the existing AdminAPI database connection and dynamically
+connects to ODS instance databases based on the `OdsInstances` table
+configuration.
+
+## Architecture
+
+### Service Layer
+
+The service files can be maintained in a common project and shared between the
+V1 and V2 projects to avoid code duplication.
+
+* `IGetEducationOrganizationQuery` - Main query handling interface
+
+* `GetEducationOrganizationQuery` - Implementation of database context query logic
+  for reading the EducationOrganizations
+
+The `RefreshEducationOrganizationCommand` service layer implements a
+comprehensive C# solution to aggregate education organization data across
+multiple ODS instances and persist it into the
+[adminapi].[EducationOrganizations] table.
+
+**Key Components:**
+
+* **Dynamic Database Connectivity**: Reads ODS instance connection strings from
+    the `OdsInstances` table and establishes connections to each ODS database
+    dynamically
```

Comment:
Might be worth noting that Admin API will need to decrypt the connection strings before using them.